connectors/src/admin/cli.ts

```diff
@@ -383,6 +389,71 @@ const slack = async (command: string, args: parseArgs.ParsedArgs) => {
 
       break;
     }
+
+    case "uninstall-for-unknown-team-ids": {
+      if (!NANGO_SLACK_CONNECTOR_ID) {
+        throw new Error("NANGO_SLACK_CONNECTOR_ID is not defined");
+      }
+
+      const slackConfigurations = await SlackConfiguration.findAll();
+      const connections = await nango_client().listConnections();
+
+      const oneHourAgo = new Date();
+      oneHourAgo.setHours(oneHourAgo.getHours() - 1);
+
+      const slackConnections = connections.connections.filter(
+        (connection: {
+          id: number;
+          connection_id: string;
+          provider: string;
+          created: string;
+        }) => {
+          const createdAt = new Date(connection.created);
+          return (
+            connection.provider === NANGO_SLACK_CONNECTOR_ID &&
+            createdAt < oneHourAgo
+          );
+        }
+      );
+
+      let connectionDetail,
+        slackTeamId: string | null = null;
+
+      const askQuestion = async (query: string): Promise<string> => {
+        const rl = readline.createInterface({
+          input: process.stdin,
+          output: process.stdout,
+        });
+
+        return new Promise((resolve) => {
+          rl.question(query, (answer) => {
+            rl.close();
+            resolve(answer);
+          });
+        });
+      };
+
+      for (const connection of slackConnections) {
+        connectionDetail = await nango_client().getConnection(
+          connection.provider,
+          connection.connection_id
+        );
+
+        slackTeamId = connectionDetail.credentials.raw.team.id;
```

Comment:
Why is slackTeamId declared in the scope above the for loop? It looks like it's not used outside of this scope.

Reply:
Same thing for `connectionDetail`

Reply:
Only to reuse the same variable throughout the loop instead of redefining new consts.

Reply:
I think this should be a const and only creates room for error. Can we move it to a const?

Reply:
Sorry, to be clearer: here we run into the potential of re-using a previous value of a non const variable where there is no reason to not use a const.
So lets use a const to limit the error surface. Especially in a code that is so sensitive and risky.

Reply:
I don't think this was dangerous and that this is necessarily a bad pattern, but since you insisted I added a commit to redefine consts inside the loop